packages/amplify-category-auth/resources/auth-custom-resource/hostedUILambda.js

```diff
@@ -1,81 +1,74 @@
 const aws = require('aws-sdk');
 const identity = new aws.CognitoIdentityServiceProvider();
-exports.handler = (event, context, callback) => {
+
+exports.handler = (event, context) => {
+  // Don't return promise, response.send() marks context as done internally
+  const ignoredPromise = handleEvent(event, context);
+};
+
+async function checkDomainAvailability(domainName) {
+  const params = { Domain: domainName };
+  try {
+    const res = await identity.describeUserPoolDomain(params).promise();
+    if (res.DomainDescription && res.DomainDescription.UserPool) {
+      return false;
+    }
+    return true;
+  } catch (err) {
+    return false;
+  }
+}
+
+async function deleteUserPoolDomain(domainName, userPoolId) {
+  const params = { Domain: domainName, UserPoolId: userPoolId };
+  await identity.deleteUserPoolDomain(params).promise();
+}
+
+async function createUserPoolDomain(domainName, userPoolId) {
+  const params = {
+    Domain: domainName,
+    UserPoolId: userPoolId,
+  };
+  await identity.createUserPoolDomain(params).promise();
+}
+
+async function handleEvent(event, context) {
   const userPoolId = event.ResourceProperties.userPoolId;
   const inputDomainName = event.ResourceProperties.hostedUIDomainName;
-  let deleteUserPoolDomain = (domainName) => {
-    let params = { Domain: domainName, UserPoolId: userPoolId };
-    return identity.deleteUserPoolDomain(params).promise();
-  };
-  if (event.RequestType == 'Delete') {
-    deleteUserPoolDomain(inputDomainName)
-      .then(() => {
-        response.send(event, context, response.SUCCESS, {});
-      })
-      .catch((err) => {
-        console.log(err);
-        response.send(event, context, response.FAILED, { err });
-      });
-  }
-  if (event.RequestType == 'Update' || event.RequestType == 'Create') {
-    let checkDomainAvailability = (domainName) => {
-      let params = { Domain: domainName };
-      return identity
-        .describeUserPoolDomain(params)
-        .promise()
-        .then((res) => {
-          if (res.DomainDescription && res.DomainDescription.UserPool) {
-            return false;
-          }
-          return true;
-        })
-        .catch((err) => {
-          return false;
-        });
-    };
-    let createUserPoolDomain = (domainName) => {
-      let params = { Domain: domainName, UserPoolId: userPoolId };
-      return identity.createUserPoolDomain(params).promise();
-    };
-    identity
-      .describeUserPool({ UserPoolId: userPoolId })
-      .promise()
-      .then((result) => {
-        if (inputDomainName) {
-          if (result.UserPool.Domain === inputDomainName) {
-            return;
+  try {
+    if (event.RequestType === 'Delete') {
+      await deleteUserPoolDomain(inputDomainName, userPoolId);
+    } else if (event.RequestType === 'Update' || event.RequestType === 'Create') {
+      const result = await identity.describeUserPool({ UserPoolId: userPoolId }).promise();
+      if (inputDomainName) {
```

Comment:
could any of these conditions be negated and early return instead? It gets really difficult to follow 3 or 4 layers deep. eg could this first condition be
```ts
if (!inputDomain) {
  response.send(event, context, response.SUCCESS, {});
  return;
}
```

Reply:
I was trying to :
- preserve code layout when moving from promises to async/await
- have single return, i.e. two `response.send` - with either success or fail.

I'll look into simplifying this christmas tree a bit more.